engine/rpcserver.go

```diff
@@ -1208,3 +1214,247 @@ func (s *RPCServer) GetAuditEvent(ctx context.Context, r *gctrpc.GetAuditEventRe
 
 	return &resp, nil
 }
+
+// GCTScriptStatus returns a slice of current running scripts that includes next run time and uuid
+func (s *RPCServer) GCTScriptStatus(ctx context.Context, r *gctrpc.GCTScriptStatusRequest) (*gctrpc.GCTScriptStatusResponse, error) {
+	if !gctscript.GCTScriptConfig.Enabled {
+		return &gctrpc.GCTScriptStatusResponse{Status: gctscript.ErrScriptingDisabled.Error()}, nil
+	}
+
+	if len(gctscript.AllVMs) < 1 {
+		return &gctrpc.GCTScriptStatusResponse{Status: "no scripts running"}, nil
+	}
+
+	resp := &gctrpc.GCTScriptStatusResponse{
+		Status: "ok",
+	}
+
+	for x := range gctscript.AllVMs {
+		resp.Scripts = append(resp.Scripts, &gctrpc.GCTScript{
+			UUID:    gctscript.AllVMs[x].ID.String(),
+			Name:    gctscript.AllVMs[x].Name,
+			NextRun: gctscript.AllVMs[x].NextRun.String(),
+		})
+	}
+
+	return resp, nil
+}
+
+func (s *RPCServer) GCTScriptQuery(ctx context.Context, r *gctrpc.GCTScriptQueryRequest) (*gctrpc.GCTScriptQueryResponse, error) {
+	if !gctscript.GCTScriptConfig.Enabled {
+		return &gctrpc.GCTScriptQueryResponse{Status: gctscript.ErrScriptingDisabled.Error()}, nil
+	}
+
+	resp := &gctrpc.GCTScriptQueryResponse{}
+
+	UUID, err := uuid.FromString(r.Script.UUID)
+	if err != nil {
+		return &gctrpc.GCTScriptQueryResponse{Status: "error"}, nil
+	}
+
+	if v, f := gctscript.AllVMs[UUID]; f {
+		resp.Status = "ok"
+		resp.Data, err = v.Read()
+		if err != nil {
+			return nil, err
+		}
+	}
+	return resp, nil
+}
+// GCTScriptExecute execute a script
+func (s *RPCServer) GCTScriptExecute(ctx context.Context, r *gctrpc.GCTScriptExecuteRequest) (*gctrpc.GCTScriptGenericResponse, error) {
+	if !gctscript.GCTScriptConfig.Enabled {
+		return &gctrpc.GCTScriptGenericResponse{Status: gctscript.ErrScriptingDisabled.Error()}, nil
+	}
+
+	if r.Script.Path == "" {
+		r.Script.Path = gctscript.ScriptPath
+	}
+
+	gctVM := gctscript.New()
+	script := filepath.Join(r.Script.Path, r.Script.Name)
+	err := gctVM.Load(script)
```

Comment:
It would be nice if we can check to see if the script name contains .gct and if it doesn't, add .gct for more user friendliness. Can either happen here or inside gctVM.Load(). We don't have to specify it in the autoload part of the config, so it would be good to keep it consistent

Reply:
I have made this change to keep it consistent across gocryptotrader, everything besides the upload cli command no longer requires a file extension as that allows you to upload either gct or zip